Jenkinsfile

```diff
@@ -326,6 +381,63 @@ pipeline {
           // FIXME can't be defined globally due to a bug in Jenkins:
           // https://issues.jenkins-ci.org/browse/JENKINS-49826
           matrixBuilder = { def matrix, int level ->
+            // Fail the entire pipeline if one step fails
+            variations.failFast = true
+            // HACK Jobs to manage resource allocation on the knl cluster
+            variations["slurm"] = {
+              try {
+                node("KNL_PRIO") { // Executor on the CoolMUC3 login node reserved for slurm allocation and management
+                  parallel "allocation": {
+                    try {
+                      timeout(time: 6, unit: 'HOURS') {
+                        // Allocate a new interactive job with up to three nodes
+                        // and two hours maximum run time. The ci-matrix will
+                        // attach subjobs to this via srun and the slurm.slurmcontrol
+                        // stage will revoke the allocation as soon as we are done.
+                        // Until that, the shell needs to stay open for the job
+                        // to survive. Hence the second "sleep 7200".
+                        sh """
+                          export SLURM_CONF=$HOME/slurm.conf
+                          salloc --job-name=mardyn-test --nodes=1-4 --partition=mpp3_batch\
+                            --tasks-per-node=3 --time=00:45:00 --begin=now+150\
+                            sleep 45m || echo 0
+                          sleep 6h
+                        """
+                      }
+                    }
+                    // Stop the slurm.allocation on any error, but do not
+                    // propatate. Needed to allow slurm.slurmcontrol (below)
+                    // to kill slurm.* by throwing an error.
+                    catch (err) {
+                      println err
+                    }
+                  }, "slurmcontrol": {
+                    // Wait for slurm.allocation to work its magic
+                    sleep 10
+                    // Store jobid
+                    knl_jobid = getKnlJobid()
+                    println "Scheduled job " + knl_jobid
+                    // Wait for all KNL jobs to finish by comparing the list
+                    // of scheduled jobs with the list of results
+                    while (results.count { key, value -> key.contains("KNL") } < variations.count { key, value -> key.contains("KNL") }) {
```

Comment:
an scancel should only be issued once all results are there, as we are running AVX code there as well